build/dangerjs/checkPerfRegressions.ts

```diff
@@ -66,7 +59,83 @@ function fluentFabricComparision(danger, markdown, warn) {
     ].join('\n'),
   )
 }
+function currentToMasterComparison(perfCounts, danger, markdown, warn) {
+  const results = _.map(
+    _.pickBy(perfCounts, value => _.has(value, 'analysis.regression')),
+    (stats, name) => {
+      const currentTicks = _.get(stats, 'analysis.numTicks')
+      const baselineTicks = _.get(stats, 'analysis.baseline.numTicks')
+
+      return {
+        name,
+        numTicks: currentTicks,
+        flamegraphFile: _.get(stats, 'processed.output.flamegraphFile'),
+        baseline: {
+          numTicks: baselineTicks,
+          flamegraphFile: _.get(stats, 'processed.output.flamegraphFile'),
+        },
+        isRegression: _.get(stats, 'analysis.regression.isRegression'),
+        currentToBaseline: Math.round((currentTicks / baselineTicks) * 100) / 100,
+      }
+    },
+  )
+
+  const regressions = _.sortBy(
+    _.filter(results, 'isRegression'),
+    stats => stats.currentToBaseline * -1,
+  )
+
+  if (regressions.length > 0) {
+    warn(`${regressions.length} perf regressions detected`)
+    markdown(
+      [
+        '## Regressions compared to master',
+        '',
+        'Scenario | Current PR Ticks | Baseline Ticks | Ratio',
+        ':--- | ---:| ---:| ---:',
+        ..._.map(regressions, (value, key) =>
+          [
+            value.name,
+            linkToFlamegraph(value.numTicks, value.flamegraphFile),
+            linkToFlamegraph(value.baseline.numTicks, value.baseline.flamegraphFile),
+            `${value.currentToBaseline}:1`,
+          ].join(' | '),
+        ),
+      ].join('\n'),
```

Comment:
Flamegrill has regression analysis built in, and it's adaptive.. it can deal with variations in overall tick count and identify new functions or functions that are consuming more time relative to total ticks.

Reply:
(It will also spell out those functions by name.)

Reply:
I know, and I am using Flamegrill's `analysis.regression.isRegression` to decide on what is a regression and what is not.

But I thought showing the ratio might also make sense, especially when improving perf. What you think?

Reply:
I'm sorry, I totally missed that you were using the feature. 

The only concern I'd have here is that would could have two tests differ by 10% and have no regression, in which case the ratio is 1.

You could also have another case where the natural variation (with no regression) contributes a 10% difference, but a function adds 5% and is marked as a regression, resulting in an overall difference a 15%. Would the ratio showing a 15% difference in that case?

In any case, I just want to affirm these are "potential" regressions and shouldn't be used as a gate. I anticipate we may have to configure or tweak thresholds for detection in flamegrill, and I still think the trend graph will be a much more reliable detector of regressions rather than the one off regression snapshots.

Reply:
We are not blocking, just warning if a potential regression is found. Let's watch the results and iterate.